File: php/features.php
<?php
declare(strict_types=1);

// Typebot SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TypebotFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TypebotBaseFeature();
            case "test":
                return new TypebotTestFeature();
            default:
                return new TypebotBaseFeature();
        }
    }
}

File: py/features.py
# Typebot SDK feature factory

from feature.base_feature import TypebotBaseFeature
from feature.test_feature import TypebotTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TypebotBaseFeature(),
        "test": lambda: TypebotTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
